Dart/src/com/jetbrains/lang/dart/ide/formatter/DartBlock.java

```diff
@@ -184,4 +221,16 @@ public BlockWithParent getParent() {
   public void setParent(BlockWithParent newParent) {
     myParent = newParent;
   }
-}
\ No newline at end of file
+
+  private static class DartCascadeBlock extends DartBlock {
```

Comment:
As I understand the question is not in programming style but in the ease of debugging.
IDE provides a lot of ways to meet the goal without changing code:
- watches
- conditional breakpoints
- log evaluated expression instead of stopping on a breakpoint
- it shows toString() as a default object presentation
- the most advanced feature is 'Java Data Type Renderers' (in Preferences): it allows to show any expression instead of toString() as a default object presentation. For example it is possible to configure that for any object of DartBlock type debugger will show getNode().getElementType().toString() + ": " + getNode().getText()